client/sites/settings/v2/index.tsx

```diff
@@ -53,10 +62,25 @@ export default function DashboardBackportSiteSettingsRenderer( {
 			return;
 		}
 
-		persistPromise.then( () => {
-			rootInstanceRef.current?.render( <Layout analyticsClient={ analyticsClient } /> );
-		} );
-	}, [ analyticsClient, siteSlug, feature ] );
+		rootInstanceRef.current?.render(
+			<Layout analyticsClient={ analyticsClient } siteSlug={ siteSlug } feature={ feature } />
+		);
+	}, [ analyticsClient, user, siteSlug, feature ] );
+
+	// Use data already available in Redux to seed the React Query cache and avoid redundant data fetching.
+	useEffect( () => {
+		if ( user ) {
+			queryClient.setQueryData( AUTH_QUERY_KEY, user );
+		}
+
+		if ( site ) {
+			queryClient.setQueryData( siteBySlugQuery( site.slug ).queryKey, site );
```

Comment:
Does this actually help? I tested and we are still waiting for the site endpoint to return before we load the Settings page 🤔 

https://github.com/user-attachments/assets/838cc294-5e9e-46e1-83dd-ad401410a46e

Or should we put this higher in the tree, like in sites-dashboard.tsx? So when the user first load the Overview page, the `siteBySlugQuery()` will be populated.